### PR TITLE
Add discriminative regularization to mlp output

### LINKED ISSUE #5: add discriminative loss for mlp and y-hat
This code has been working great for me. Currently, the discriminative regularization in this codebase uses the loss from the batch norm layers of the ConvolutionalSequence, but the paper covers the more general case of taking loss from elsewhere in the classifier, as shown below:

4654ee39-75d4-463c-b808-8bd4a2c5e616

I'd like to add losses at the higher layers - MLP and y-hat. I'm curious if anyone else is interested in working on this with me - the discriminative regularization parts are some of trickier parts of building the training computation graph, and y-hat currently doesn't have a named node in the classifier graph.

### PR DESCRIPTION
**Pull request by dribnet**
Mostly resolves #5.

@vdumoulin - curious if this looks roughly right. I've confirmed that training starts, but haven't fully trained a model with this change yet. Can do so if this doesn't have any obvious flaws and that would be useful.

Note: also includes a commit to change model loading so that this can be run with more recent vintage blocks.

### CODE REVIEW

**Comment by dribnet** [comment]
Update: this runs and trains, but the loss explodes after about 6 epochs - and I've confirmed it's the new discriminative mlp term causing this. So something here is not quite right.
